src/microbe_stage/MembraneShapeGenerator.cs

```diff
@@ -144,42 +149,42 @@ public MembranePointData GenerateShape(IMembraneDataSource parameters)
         return (mesh, surfaceIndex);
     }
 
-    private static int InitializeCorrectMembrane(Vector2[] vertices2D, int vertexCount, int writeIndex,
-        Vector3[] vertices, Vector2[] uvs, MembraneType membraneType)
+    /// <summary>
+    ///   Takes a mesh with placed vertices and places UVs and normals on it
+    /// </summary>
+    private static void FinishMesh(int vertexCount, int layerCount, Vector3[] vertices, Vector2[] uvs,
+        Vector3[] normals)
     {
-        // common variables
-        float height = 0.1f;
-        float multiplier = 2.0f * MathF.PI;
-        var center = new Vector2(0.5f, 0.5f);
-
-        // cell walls need obvious inner/outer membranes (we can worry
-        // about chitin later)
-        if (membraneType.CellWall)
+        for (int layer = 0; layer < layerCount; ++layer)
         {
-            height = 0.05f;
-        }
+            for (int i = 0; i < vertexCount; ++i)
+            {
+                int id = i + layer * vertexCount;
+                float turnFraction = 2.0f * MathF.PI * i / vertexCount;
```

Comment:
Would pre-calculating `2 * PI / vertexCount` be faster? (as that would avoid quite many divisions and multiplies per loop iteration)